### ISSUE TITLE
Http proxy

### ISSUE THREAD

**Issue by typicalnick** [LonamiWebs/Telethon#477]
Have a timout problem with client.connect()

Versions:
ubuntu 16.04
python 3.5.2
telethon 0.15.5.7

Code:
api_id = XXXXXX
api_hash = '********************************'
phone = '+XXXXXXXXXXX'

host = 'XXX.XX.X.XXX'     # valid proxy
port = XXXX                    # valid port
proxy = (socks.HTTP, host, port)

client = TelegramClient('test', api_id, api_hash, proxy, timeout=timedelta(seconds=30))
client.connect()

gives me "socket.timeout: timed out". I have tried proxy=None also, since $http_proxy is defined in linux environments (package "requests" work perfectly with it). And I had the same error. 

Do you have idea how to check or fix it?
Thanks

**Comment by Lonami** [comment]
> Do you have idea how to check or fix it?

Not really.

**Comment by typicalnick** [comment]
Yeah, i read it in other issues) Nevertheless many thanks for your answer. I forgot to write i have ipv4 proxy. And i'll be happy for any help.

**Comment by qiaofei32** [comment]
`socks.set_default_proxy(socks.SOCKS5, addr="localhost", port=1080)
socket.socket = socks.socksocket
socket.setdefaulttimeout(2.0)
api_id = xxxx
api_hash = 'xxxx'
client = TelegramClient('xxx', api_id, api_hash)
client.start()`

**Comment by kaxap** [comment]
Just tested this code and it works well

```python3
proxy = (socks.HTTP, proxy_host, proxy_port,)
client = TelegramClient(phone_number, api_id, api_hash, proxy=proxy)
client.connect()
```

**Comment by steven1227** [comment]
@Lonami when I try the ConntectionHttp , it says `missing 1 required keyword-only argument: 'loop'`

**Comment by Lonami** [comment]
Version? Traceback?

**Comment by steven1227** [comment]
@Lonami I use the telethon-sync version

**Comment by Lonami** [comment]
Yeah right I just pushed a fix.
